### ISSUE TITLE
`consolidate_intersections` causes kernel restart when pygeos is installed

### ISSUE THREAD

**Issue by brl0** [gboeing/osmnx#681]
First of all, I really appreciate this awesome project!

I ran into an issue that only occurs when pygeos is installed.

**Problem description**
  - What did you do?
Using `movingpandas` [example](https://mybinder.org/v2/gh/anitagraser/movingpandas-examples/main?filepath=analysis-examples/4-stop-hotspots.ipynb) runs successfully in binder without `pygeos` installed.
After installing `pygeos`, kernel restarts when calling `consolidate_intersections`.
Tried on binder, locally, and cloud based jupyter instance in python 3.7 and 3.8.  

  - What did you expect to happen?
Expected to work as it does without `pygeos`.
  - And what actually happened instead?
Kernel restarts.

**Environment information**
  - What operating system are you using?
Linux
  - What Python version are you using?
3.7.10, 3.8.8
  - What OSMnx version are you using?
1.0.1
  - Provide a complete list of your environment's packages and their versions (for example, run `conda list` or `pip list` then paste the output between the two "details" tags below)

<details>

```
# packages in environment at /srv/conda/envs/notebook:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                       1_gnu    conda-forge
affine                    2.3.0                      py_0    conda-forge
alembic                   1.5.5              pyhd8ed1ab_0    conda-forge
anyio                     2.1.0            py37h89c1867_0    conda-forge
argon2-cffi               20.1.0           py37h8f50634_2    conda-forge
async_generator           1.10                       py_0    conda-forge
attrs                     20.3.0             pyhd3deb0d_0    conda-forge
babel                     2.9.0              pyhd3deb0d_0    conda-forge
backcall                  0.2.0              pyh9f0ad1d_0    conda-forge
backports                 1.0                        py_2    conda-forge
backports.functools_lru_cache 1.6.1                      py_0    conda-forge
beautifulsoup4            4.9.3              pyhb0f4dca_0    conda-forge
bleach                    3.3.0              pyh44b312d_0    conda-forge
blinker                   1.4                        py_1    conda-forge
blosc                     1.21.0               h9c3ff4c_0    conda-forge
bokeh                     2.3.0            py37h89c1867_0    conda-forge
boost-cpp                 1.72.0               h9d3c048_4    conda-forge
branca                    0.4.2              pyhd8ed1ab_0    conda-forge
brotlipy                  0.7.0           py37hb5d75c8_1001    conda-forge
bs4                       4.9.3                         0    conda-forge
bzip2                     1.0.8                h7f98852_4    conda-forge
c-ares                    1.17.1               h36c2ea0_0    conda-forge
ca-certificates           2020.12.5            ha878542_0    conda-forge
cairo                     1.16.0            h7979940_1007    conda-forge
cartopy                   0.18.0          py37h26456f9_13    conda-forge
certifi                   2020.12.5        py37h89c1867_1    conda-forge
certipy                   0.1.3                      py_0    conda-forge
cffi                      1.14.5           py37hc58025e_0    conda-forge
cfitsio                   3.470                hb418390_7    conda-forge
cftime                    1.4.1            py37h902c9e0_0    conda-forge
chardet                   4.0.0            py37h89c1867_1    conda-forge
click                     7.1.2              pyh9f0ad1d_0    conda-forge
click-plugins             1.1.1                      py_0    conda-forge
cligj                     0.7.1              pyhd8ed1ab_0    conda-forge
cloudpickle               1.6.0                      py_0    conda-forge
colorcet                  2.0.6              pyhd8ed1ab_0    conda-forge
cryptography              3.4.4            py37hf1a17b8_0    conda-forge
curl                      7.71.1               he644dc0_8    conda-forge
cycler                    0.10.0                     py_2    conda-forge
cython                    0.29.22          py37hcd2ae1e_0    conda-forge
cytoolz                   0.11.0           py37h5e8e339_3    conda-forge
dask                      2021.3.0           pyhd8ed1ab_0    conda-forge
dask-core                 2021.3.0           pyhd8ed1ab_0    conda-forge
datashader                0.12.1             pyh44b312d_0    conda-forge
datashape                 0.5.4                      py_1    conda-forge
dbus                      1.13.18              hb2f20db_0    defaults
decorator                 4.4.2                      py_0    conda-forge
defusedxml                0.6.0                      py_0    conda-forge
descartes                 1.1.0                      py_4    conda-forge
distributed               2021.3.0         py37h89c1867_0    conda-forge
entrypoints               0.3             py37hc8dfbb8_1002    conda-forge
esda                      2.3.6              pyhd8ed1ab_0    conda-forge
expat                     2.2.10               h9c3ff4c_0    conda-forge
fiona                     1.8.18           py37h17d6ad9_1    conda-forge
folium                    0.12.0             pyhd8ed1ab_1    conda-forge
fontconfig                2.13.1            hba837de_1004    conda-forge
freetype                  2.10.4               h0708190_1    conda-forge
freexl                    1.0.6                h7f98852_0    conda-forge
fsspec                    0.8.7              pyhd8ed1ab_0    conda-forge
gdal                      3.2.1            py37hc5bc4e4_7    conda-forge
geographiclib             1.50                       py_0    conda-forge
geopandas                 0.9.0              pyhd8ed1ab_0    conda-forge
geopy                     2.1.0              pyhd3deb0d_0    conda-forge
geos                      3.9.1                h9c3ff4c_2    conda-forge
geotiff                   1.6.0                h11d48b3_4    conda-forge
geoviews                  1.9.1              pyhd8ed1ab_0    conda-forge
geoviews-core             1.9.1              pyha770c72_0    conda-forge
gettext                   0.19.8.1          h0b5b191_1005    conda-forge
giddy                     2.3.3                      py_0    conda-forge
giflib                    5.2.1                h516909a_2    conda-forge
glib                      2.68.0               h9c3ff4c_1    conda-forge
glib-tools                2.68.0               h9c3ff4c_1    conda-forge
gmp                       6.2.1                h58526e2_0    conda-forge
gmpy2                     2.1.0b1          py37hcb968a4_1    conda-forge
gst-plugins-base          1.18.4               h29181c9_0    conda-forge
gstreamer                 1.18.4               h76c114f_0    conda-forge
hdf4                      4.2.13            h10796ff_1004    conda-forge
hdf5                      1.10.6          nompi_h6a2412b_1114    conda-forge
heapdict                  1.0.1                      py_0    conda-forge
holoviews                 1.14.2             pyhd8ed1ab_0    conda-forge
hvplot                    0.7.1              pyh44b312d_0    conda-forge
icu                       68.1                 h58526e2_0    conda-forge
idna                      2.10               pyh9f0ad1d_0    conda-forge
importlib-metadata        3.4.0            py37h89c1867_0    conda-forge
importlib_metadata        3.4.0                hd8ed1ab_0    conda-forge
ipykernel                 5.5.0            py37h888b3d9_1    conda-forge
ipython                   7.20.0           py37h888b3d9_2    conda-forge
ipython_genutils          0.2.0                      py_1    conda-forge
ipywidgets                7.6.3              pyhd3deb0d_0    conda-forge
jedi                      0.18.0           py37h89c1867_2    conda-forge
jinja2                    2.11.3             pyh44b312d_0    conda-forge
joblib                    1.0.1              pyhd8ed1ab_0    conda-forge
jpeg                      9d                   h516909a_0    conda-forge
json-c                    0.15                 h98cffda_0    conda-forge
json5                     0.9.5              pyh9f0ad1d_0    conda-forge
jsonschema                3.2.0            py37hc8dfbb8_1    conda-forge
jupyter-offlinenotebook   0.2.1              pyhd8ed1ab_0    conda-forge
jupyter-resource-usage    0.5.1              pyhd8ed1ab_0    conda-forge
jupyter_client            6.1.11             pyhd8ed1ab_1    conda-forge
jupyter_core              4.7.1            py37h89c1867_0    conda-forge
jupyter_server            1.4.0            py37h89c1867_0    conda-forge
jupyter_telemetry         0.1.0              pyhd8ed1ab_1    conda-forge
jupyterhub-base           1.3.0            py37h89c1867_0    conda-forge
jupyterhub-singleuser     1.3.0            py37h89c1867_0    conda-forge
jupyterlab                3.0.7              pyhd8ed1ab_0    conda-forge
jupyterlab_pygments       0.1.2              pyh9f0ad1d_0    conda-forge
jupyterlab_server         2.3.0              pyhd8ed1ab_0    conda-forge
jupyterlab_widgets        1.0.0              pyhd8ed1ab_1    conda-forge
kealib                    1.4.14               hcc255d8_2    conda-forge
kiwisolver                1.3.1            py37h2527ec5_1    conda-forge
krb5                      1.17.2               h926e7f8_0    conda-forge
lcms2                     2.12                 hddcbb42_0    conda-forge
ld_impl_linux-64          2.35.1               hea4e1c9_2    conda-forge
libblas                   3.9.0                8_openblas    conda-forge
libcblas                  3.9.0                8_openblas    conda-forge
libclang                  11.1.0          default_ha53f305_0    conda-forge
libcurl                   7.71.1               hcdd3856_8    conda-forge
libdap4                   3.20.6               hd7c4107_1    conda-forge
libedit                   3.1.20191231         he28a2e2_2    conda-forge
libev                     4.33                 h516909a_1    conda-forge
libevent                  2.1.10               hcdb4288_3    conda-forge
libffi                    3.3                  h58526e2_2    conda-forge
libgcc-ng                 9.3.0               h2828fa1_18    conda-forge
libgdal                   3.2.1                h38ff51b_7    conda-forge
libgfortran-ng            9.3.0               hff62375_18    conda-forge
libgfortran5              9.3.0               hff62375_18    conda-forge
libglib                   2.68.0               h3e27bee_1    conda-forge
libgomp                   9.3.0               h2828fa1_18    conda-forge
libiconv                  1.16                 h516909a_0    conda-forge
libkml                    1.3.0             hd79254b_1012    conda-forge
liblapack                 3.9.0                8_openblas    conda-forge
libllvm10                 10.0.1               he513fc3_3    conda-forge
libllvm11                 11.1.0               hf817b99_0    conda-forge
libnetcdf                 4.7.4           nompi_h56d31a8_107    conda-forge
libnghttp2                1.43.0               h812cca2_0    conda-forge
libopenblas               0.3.12          pthreads_h4812303_1    conda-forge
libpng                    1.6.37               hed695b0_2    conda-forge
libpq                     13.1                 hfd2b0eb_2    conda-forge
libpysal                  4.4.0              pyhd8ed1ab_0    conda-forge
librttopo                 1.1.0                h1185371_6    conda-forge
libsodium                 1.0.18               h516909a_1    conda-forge
libspatialindex           1.9.3                he1b5a44_3    conda-forge
libspatialite             5.0.1                he52d314_3    conda-forge
libssh2                   1.9.0                hab1572f_5    conda-forge
libstdcxx-ng              9.3.0               h6de172a_18    conda-forge
libtiff                   4.2.0                hdc55705_0    conda-forge
libuuid                   2.32.1            h14c3975_1000    conda-forge
libwebp-base              1.2.0                h7f98852_2    conda-forge
libxcb                    1.14                 h7b6447c_0    defaults
libxkbcommon              1.0.3                he3ba5ed_0    conda-forge
libxml2                   2.9.10               h72842e0_3    conda-forge
llvmlite                  0.36.0           py37h9d7f4d0_0    conda-forge
locket                    0.2.1            py37h06a4308_1    defaults
lz4-c                     1.9.3                h9c3ff4c_0    conda-forge
lzo                       2.10              h516909a_1000    conda-forge
mako                      1.1.4              pyh44b312d_0    conda-forge
mapclassify               2.4.2              pyhd8ed1ab_0    conda-forge
markdown                  3.3.4              pyhd8ed1ab_0    conda-forge
markupsafe                1.1.1            py37h5e8e339_3    conda-forge
matplotlib                3.3.4            py37h89c1867_0    conda-forge
matplotlib-base           3.3.4            py37h0c9df89_0    conda-forge
mistune                   0.8.4           py37h5e8e339_1003    conda-forge
mock                      4.0.3            py37h89c1867_1    conda-forge
movingpandas              0.5rc1                     py_0    conda-forge
mpc                       1.1.0             h04dde30_1009    conda-forge
mpfr                      4.0.2                he80fd80_1    conda-forge
mpmath                    1.2.1              pyhd8ed1ab_0    conda-forge
msgpack-python            1.0.2            py37h2527ec5_1    conda-forge
multipledispatch          0.6.0                      py_0    conda-forge
munch                     2.5.0                      py_0    conda-forge
mysql-common              8.0.23               ha770c72_1    conda-forge
mysql-libs                8.0.23               h935591d_1    conda-forge
nbclassic                 0.2.6              pyhd8ed1ab_0    conda-forge
nbclient                  0.5.2              pyhd8ed1ab_0    conda-forge
nbconvert                 6.0.7            py37h89c1867_3    conda-forge
nbformat                  5.1.2              pyhd8ed1ab_1    conda-forge
ncurses                   6.2                  h58526e2_4    conda-forge
nest-asyncio              1.4.3              pyhd8ed1ab_0    conda-forge
netcdf4                   1.5.6           nompi_py37hdbee05a_100    conda-forge
networkx                  2.5                        py_0    conda-forge
nose                      1.3.7           py37hc8dfbb8_1004    conda-forge
notebook                  6.2.0            py37h89c1867_0    conda-forge
nspr                      4.30                 h9c3ff4c_0    conda-forge
nss                       3.63                 hb5efdd6_0    conda-forge
nteract_on_jupyter        2.1.3                      py_0    conda-forge
numba                     0.53.0           py37h7dd73a4_1    conda-forge
numexpr                   2.7.3            py37hdc94413_0    conda-forge
numpy                     1.20.1           py37haa41c4c_0    conda-forge
oauthlib                  3.0.1                      py_0    conda-forge
olefile                   0.46               pyh9f0ad1d_1    conda-forge
openjpeg                  2.4.0                hf7af979_0    conda-forge
openssl                   1.1.1j               h7f98852_0    conda-forge
osmnx                     1.0.1              pyhd3deb0d_0    conda-forge
packaging                 20.9               pyh44b312d_0    conda-forge
pamela                    1.0.0                      py_0    conda-forge
pandas                    1.2.3            py37hdc94413_0    conda-forge
pandoc                    2.11.4               h7f98852_0    conda-forge
pandocfilters             1.4.2                      py_1    conda-forge
panel                     0.11.1             pyhd8ed1ab_0    conda-forge
param                     1.10.1             pyhd3deb0d_0    conda-forge
parso                     0.8.1              pyhd8ed1ab_0    conda-forge
partd                     1.1.0                      py_0    conda-forge
patsy                     0.5.1                      py_0    conda-forge
pcre                      8.44                 he1b5a44_0    conda-forge
pexpect                   4.8.0            py37hc8dfbb8_1    conda-forge
pickleshare               0.7.5           py37hc8dfbb8_1002    conda-forge
pillow                    8.1.2            py37h4600e1f_0    conda-forge
pip                       21.0.1             pyhd8ed1ab_0    conda-forge
pixman                    0.40.0               h36c2ea0_0    conda-forge
poppler                   0.89.0               h2de54a5_5    conda-forge
poppler-data              0.4.10                        0    conda-forge
postgresql                13.1                 h6303168_2    conda-forge
proj                      7.2.0                h277dcde_2    conda-forge
prometheus_client         0.9.0              pyhd3deb0d_0    conda-forge
prompt-toolkit            3.0.16             pyha770c72_0    conda-forge
psutil                    5.8.0            py37h5e8e339_1    conda-forge
ptyprocess                0.7.0              pyhd3deb0d_0    conda-forge
pycparser                 2.20               pyh9f0ad1d_2    conda-forge
pyct                      0.4.8                    py37_0    defaults
pycurl                    7.43.0.6         py37h88a64d2_1    conda-forge
pygments                  2.8.0              pyhd8ed1ab_0    conda-forge
pyjwt                     2.0.1              pyhd8ed1ab_0    conda-forge
pyopenssl                 20.0.1             pyhd8ed1ab_0    conda-forge
pyparsing                 2.4.7              pyh9f0ad1d_0    conda-forge
pyproj                    3.0.1            py37h63c117d_0    conda-forge
pyqt                      5.12.3           py37h89c1867_7    conda-forge
pyqt-impl                 5.12.3           py37he336c9b_7    conda-forge
pyqt5-sip                 4.19.18          py37hcd2ae1e_7    conda-forge
pyqtchart                 5.12             py37he336c9b_7    conda-forge
pyqtwebengine             5.12.1           py37he336c9b_7    conda-forge
pyrsistent                0.17.3           py37h5e8e339_2    conda-forge
pyshp                     2.1.3              pyh44b312d_0    conda-forge
pysocks                   1.7.1            py37h89c1867_3    conda-forge
pytables                  3.6.1            py37he17a9a8_3    conda-forge
python                    3.7.10          hffdb5ce_100_cpython    conda-forge
python-dateutil           2.8.1                      py_0    conda-forge
python-editor             1.0.4                      py_0    conda-forge
python-json-logger        2.0.1              pyh9f0ad1d_0    conda-forge
python_abi                3.7                     1_cp37m    conda-forge
pytz                      2021.1             pyhd8ed1ab_0    conda-forge
pyviz_comms               2.0.1              pyhd3deb0d_0    conda-forge
pyyaml                    5.4.1            py37h5e8e339_0    conda-forge
pyzmq                     22.0.3           py37h499b945_0    conda-forge
qt                        5.12.9               hda022c4_4    conda-forge
quantecon                 0.4.8            py37hc8dfbb8_0    conda-forge
rasterio                  1.2.1            py37hb284c01_2    conda-forge
readline                  8.0                  he28a2e2_2    conda-forge
requests                  2.25.1             pyhd3deb0d_0    conda-forge
rtree                     0.9.7            py37h0b55af0_1    conda-forge
ruamel.yaml               0.16.12          py37h5e8e339_2    conda-forge
ruamel.yaml.clib          0.2.2            py37h5e8e339_2    conda-forge
scikit-learn              0.24.1           py37h69acf81_0    conda-forge
scipy                     1.6.1            py37h14a347d_0    conda-forge
seaborn                   0.11.1               ha770c72_0    conda-forge
seaborn-base              0.11.1             pyhd8ed1ab_1    conda-forge
send2trash                1.5.0                      py_0    conda-forge
setuptools                49.6.0           py37h89c1867_3    conda-forge
shapely                   1.7.1            py37hf7ed6d2_4    conda-forge
six                       1.15.0             pyh9f0ad1d_0    conda-forge
sniffio                   1.2.0            py37h89c1867_1    conda-forge
snuggs                    1.4.7                      py_0    conda-forge
sortedcontainers          2.3.0              pyhd8ed1ab_0    conda-forge
soupsieve                 2.2                pyhd3eb1b0_0    defaults
spaghetti                 1.5.6              pyhd8ed1ab_0    conda-forge
splot                     1.1.3              pyhd8ed1ab_1    conda-forge
spreg                     1.2.2              pyhd8ed1ab_0    conda-forge
sqlalchemy                1.3.23           py37h5e8e339_0    conda-forge
sqlite                    3.34.0               h74cdb3f_0    conda-forge
statsmodels               0.12.2           py37h902c9e0_0    conda-forge
sympy                     1.7.1            py37h89c1867_1    conda-forge
tblib                     1.7.0              pyhd8ed1ab_0    conda-forge
terminado                 0.9.2            py37h89c1867_0    conda-forge
testpath                  0.4.4                      py_0    conda-forge
threadpoolctl             2.1.0              pyh5ca1d4c_0    conda-forge
tiledb                    2.2.6                h91fcb0e_0    conda-forge
tk                        8.6.10               hed695b0_1    conda-forge
toolz                     0.11.1                     py_0    conda-forge
tornado                   6.1              py37h5e8e339_1    conda-forge
tqdm                      4.59.0             pyhd8ed1ab_0    conda-forge
traitlets                 5.0.5                      py_0    conda-forge
typing_extensions         3.7.4.3                    py_0    conda-forge
tzcode                    2021a                h7f98852_1    conda-forge
tzdata                    2021a                he74cb21_0    conda-forge
urllib3                   1.26.3             pyhd8ed1ab_0    conda-forge
wcwidth                   0.2.5              pyh9f0ad1d_2    conda-forge
webencodings              0.5.1                      py_1    conda-forge
wheel                     0.36.2             pyhd3deb0d_0    conda-forge
widgetsnbextension        3.5.1            py37hc8dfbb8_4    conda-forge
xarray                    0.17.0             pyhd8ed1ab_0    conda-forge
xerces-c                  3.2.3                h9d8b166_2    conda-forge
xorg-kbproto              1.0.7             h14c3975_1002    conda-forge
xorg-libice               1.0.10               h516909a_0    conda-forge
xorg-libsm                1.2.3             hd9c2040_1000    conda-forge
xorg-libx11               1.7.0                h36c2ea0_0    conda-forge
xorg-libxext              1.3.4                h7f98852_1    conda-forge
xorg-libxrender           0.9.10            h7f98852_1003    conda-forge
xorg-renderproto          0.11.1            h14c3975_1002    conda-forge
xorg-xextproto            7.3.0             h14c3975_1002    conda-forge
xorg-xproto               7.0.31            h14c3975_1007    conda-forge
xz                        5.2.5                h516909a_1    conda-forge
yaml                      0.2.5                h516909a_0    conda-forge
zeromq                    4.3.4                h9c3ff4c_0    conda-forge
zict                      2.0.0                      py_0    conda-forge
zipp                      3.4.0                      py_0    conda-forge
zlib                      1.2.11            h516909a_1010    conda-forge
zstd                      1.4.9                ha95c52a_0    conda-forge
```

</details>

**Provide a complete minimal reproducible example**

```python
import osmnx as ox

buffer = 0.005
minx, miny, maxx, maxy = 116.371017, 39.859827, 116.411643, 39.899299

G = ox.graph_from_bbox(maxy+buffer, miny-buffer, minx-buffer, maxx+buffer, network_type='all')
G_proj = ox.project_graph(G)
G2 = ox.consolidate_intersections(G_proj, rebuild_graph=True, tolerance=15, dead_ends=False)
```

**Comment by gboeing** [comment]
This should have been fixed in #679. My diagnosis was that `shapely.ops` functions do not play nicely with `pygeos` geometry objects right now.

Would you try the master branch and see if it's fixed for you?

**Comment by brl0** [comment]
That worked, thanks for your response, and thanks again for the great project.
